### DISCUSSION TITLE
Ability to select help for controls/windows

### DISCUSSION THREAD

**Discussion by nvdaes** [nvdaes/controlUsageAssistant#10] [Ideas]
This has been requested in this [NVDA issue](https://github.com/nvaccess/nvda/issues/13977).

We have different options. For example:

- Add an option to configure help text based just on roles.
- Option to configure help text based on roles and Python class mro.
- Option based on title and role.
- Option based on states, for instance, if a checkbox is checked or not.
- Etc.

If this is applicable to automatic messages, a single line edit box should suffice. Otherwise, we may use multiline.

Before creating a pull request, I'd like to seek for community feedback.

**Comment by LittleStar-VIP** [comment]
Hello,
I would like to know currently how would this function deal with different locale?
say for example I have a set of help text set under tw locale, then i switch to another chinese locale, or even english locale, what will happen?

**Comment by nvdaes** [comment]
Hello:

Since it would be configurable for different profiles, you may set
different messages for different locales and switch to the appropriate
profile, for example assigning keystrokes in input gestures dialog.
Let us know if this wouldn't be enough for you

2022-08-04 8:07 GMT+02:00, LITTLE STAR☆VIP ***@***.***>:

**Comment by ruifontes** [comment]
I agree with the need, for some users, of help messages...
For me, they should have three levels:
0 - No message is announced;
1 - A brief message is announced;
2 - A normal message is announced.

Regarding locale changes, for me the messages should be localizables, and so, if the user changes the locale, by instance from chinese to english, the messages will be announced using the english localized version...

**Comment by nvdaes** [comment]
I think that the current option to provide messages with NVDA+h, and automatic messages that can be deactivated provides different levels of info and messages are translatable.
Here we can think about the possibility of custom messages that is, an edit box where users or trainers can set the message to be spoken when certain control or Windows is focused, so translators cannot work in these kind of messages since they would be configured for each person, once the add-on has been released. Of course, users can set messages for different languages in different profiles.
Comments are appreciated.
